src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -853,9 +876,17 @@ MultiClientRegisterWait(WaitInfo *waitInfo, TaskExecutionStatus executionStatus,
 						int32 connectionId)
 {
 	MultiConnection *connection = NULL;
+#ifdef HAVE_POLL
 	struct pollfd *pollfd = NULL;
+#else
+	int connectionFileDescriptor = 0;
+#endif
 
-	Assert(waitInfo->registeredWaiters < waitInfo->maxWaiters);
+	/* This is to make sure we could never register more than maxWaiters in Windows */
+	if (waitInfo->registeredWaiters >= waitInfo->maxWaiters)
+	{
+		return;
+	}
```

Comment:
What would happen if a job needs more connections than `maxWaiters`? We either shouldn't allow more than `maxWaiters` connections by erroring out, or be sure that we can handle it.

Reply:
In that case we'll have worse-than-usual performance. We'll wait until one of the `maxWaiters` connections becomes ready. When a connection outside of that set becomes ready we won't react until one of the other connections triggers. Since we go down the list and try to use every connection whenever one of them has an event I don't expect that this would have a noticeable impact on performance.